rest/config.go

```diff
@@ -1463,8 +1463,24 @@ func (sc *ServerContext) fetchDatabase(ctx context.Context, dbName string) (foun
 		// inherit properties the bootstrap config
 		cnf.CACertPath = sc.Config.Bootstrap.CACertPath
 
+		// We need to check for corruption in the database config (CC. CBG-3292). If the fetched config doesn't match the
+		// bucket name we got the config from we need to maker this db context as corrupt. Then remove the context and
+		// in memory representation on the server context.
 		bucketCopy := bucket
-		cnf.Bucket = &bucketCopy
+		if bucket != *cnf.Bucket {
+			base.DebugfCtx(ctx, base.KeyConfig, "Mismatch in database config bucket name: %s and backend bucket: %s You must update database config immediately", *cnf.Bucket, bucket)
```

Comment:
We almost certainly want to log this as a warning here so it doesn't rotate out of the logs, since warning logs have a very low log rotation period.